### PR TITLE
Add support for snowflake dialect

### PR DESCRIPTION
**Pull request by scgkiran**
Steps to run snowflake dialect tests.

1. Update `bft/testers/snowflake/config.yaml` file with snowflake `user,account,db,schema` etc. 
2. `export SNOWSQL_PWD=<your password>` 
3. To run the snowflake tests `pytest bft/tests/test_snowflake.py`

### CODE REVIEW

File: bft/testers/snowflake/runner.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                            password=os.environ['SNOWSQL_PWD'],
+                            account=sf_config['account'],
+                            database=sf_config['database'],
+                            schema=sf_config['schema'],
+                            # host=sf_config['hostname'].get(""),
+                            # role=sf_config['role'].get(""),
+                            warehouse=sf_config['warehouse']
+                            )
+
+    def run_sql_case(self, case: Case, mapping: SqlMapping) -> SqlCaseResult:
```
Comment (jacques-n) [inline bft/testers/snowflake/runner.py:86]: i'm guessing a large portion of this is duplicated between here and postgres. We should figure out how to factor out and share.

File: bft/testers/snowflake/runner.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+from snowflake.connector import connect
+from snowflake.connector.errors import Error
+
+from bft.cases.runner import SqlCaseResult, SqlCaseRunner
+from bft.cases.types import Case
+from bft.dialects.types import SqlMapping
+
+type_map = {
+    "i8": "TINYINT",
```
Comment (jacques-n) [inline bft/testers/snowflake/runner.py]: Aren't most of these shorter numeric just synonyms for decimal(38,0)? If so I suspect we shouldn't be binding them to smaller types. For example, I think i32+i32 output decimal(38,0) in snowflake as opposed to what is likely defined in substrait where i32 + i32 => i32.

File: bft/testers/snowflake/runner.py (hunk 1 above)
Comment (scgkiran) [reply bft/testers/snowflake/runner.py:86]: I want to do this refactoring at a later point in time. Just want to see other changes that will be needed in snowflake tester.

File: bft/testers/snowflake/runner.py (hunk 2 above)
Comment (scgkiran) [reply bft/testers/snowflake/runner.py]: Made changes in current PR to make these types as unsupported. 
For functions with Decimal types, I need to address two issues.
1. Handle duplicate function name in BFT.
2. Add missing functions with Decimal to Substrait.

**Comment by scgkiran** [comment]
Closing this as I opened [PR](https://github.com/voltrondata/bft/pull/76) on the volrtondata/bft repo

**Pull request closed without merging**